### ISSUE TITLE
subtract background

### ISSUE THREAD

**Issue by haviduck** [JizhiziLi/GFM#18]
heya, thanks for sharing this, gave me alot of help trying to figure out segmentation. ive worked with watershed and contours earlier but a novice at best. wondering if you could push me in the right direction for enhancing the background matting as its picking up some noise (ref image). it clearly finds the fish and is really close to closing that mask, 
i tried hough but no luck.  

![IMG_20210124_211754](https://user-images.githubusercontent.com/4920887/105642472-386b8380-5e8a-11eb-952b-e29610948214.png)

**Comment by JizhiziLi** [comment]
Hi there, it seems like the model fails to extract the accurate semantic information in this picture, i.e. the output from the glance decoder. Please note that the pre-trained model is designed to process a high-resolution image (typically 1K times 2K), so might fail on a low-resolution image.

One alternative solution you can try by using our pre-trained model is changing the testing strategy, as referred to **section 6.3** in our paper. Currently, we are using hybrid-resolution test strategy with ratios set as 1/2, 1/3. You can try different ratios or use the same ratio for two decoders to see if you can get better results.

Besides this, it might be better if you can re-train the network on your target dataset, I believe you can get better results than our pre-trained model. : )

**Comment by haviduck** [comment]
> Hi there, it seems like the model fails to extract the accurate semantic information in this picture, i.e. the output from the glance decoder. Please note that the pre-trained model is designed to process a high-resolution image (typically 1K times 2K), so might fail on a low-resolution image.
> 
> One alternative solution you can try by using our pre-trained model is changing the testing strategy, as referred to **section 6.3** in our paper. Currently, we are using hybrid-resolution test strategy with ratios set as 1/2, 1/3. You can try different ratios or use the same ratio for two decoders to see if you can get better results.
> 
> Besides this, it might be better if you can re-train the network on your target dataset, I believe you can get better results than our pre-trained model. : )

Thank you,  i am using superres to upscale the original photo (what i shared was a roi i manually cut out). ill try first to upscale the roi itself, sharpen it and then see if it works better :) if not ill retrain! hope you dont mind if i ask some questions underway if i get stuck. promise to thoroughly test and try before that.

**Comment by JizhiziLi** [comment]
No worries, let me know if you have further questions!

**Comment by haviduck** [comment]
sure. training code will be released after your paper review?

**Comment by JizhiziLi** [comment]
Yes along with the AM-2k dataset.
